backend/src/main.cpp

```diff
@@ -205,9 +205,15 @@ Napi::Boolean SetFrequencyState(const Napi::CallbackInfo& info)
 
     mClient->SetOnHeadset(frequency, !onSpeaker);
 
+    // Included for legacy reasons in case some older client depends on this message.
     MainThreadShared::mApiServer->handleAFVEventForWebsocket(
         sdk::types::Event::kFrequencyStateUpdate, {}, {});
 
+    // New event that only notifies of the change to this specific station.
+    auto stateJson = MainThreadShared::mApiServer->buildStationStateJson(std::nullopt, frequency);
+    MainThreadShared::mApiServer->publishStationState(stateJson);
+    NapiHelpers::callElectron("station-state-update", stateJson.dump());
```

Comment:
This causes everyone to update, both websocket connections and the UI, whenever something is changed via IPC message.

Technically the electron station-state-update isn't necessary right now and could be omitted, but I put it in to make this have the same update message behaviour as updates caused by an incoming websocket message. More on that later...